language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -559,7 +560,7 @@ modVar prefix parts = prefix <> T.intercalate "_" parts
 -- Calculate a filepath from a module name e.g. 'modPath [".", "A",
 -- "B"]' is "./A/B".
 modPath :: [T.Text] -> T.Text
-modPath parts = T.pack (joinPath $ map T.unpack parts)
+modPath parts = T.intercalate "/" parts
```

Comment:
Given that you already import `System.FilePath.Posix`, maybe just use `joinPath` from there?

Reply:
we could, i like that this approach allows us to avoid unpacks.

Reply:
Makes sense :+1: